x/mongo/driver/topology/errors.go

```diff
@@ -95,14 +102,19 @@ func (w WaitQueueTimeoutError) Error() string {
 		)
 	}
 
-	return fmt.Sprintf(
-		"%s; maxPoolSize: %d, connections in use by cursors: %d"+
-			", connections in use by transactions: %d, connections in use by other operations: %d",
-		errorMsg,
-		w.maxPoolSize,
-		w.PinnedCursorConnections,
-		w.PinnedTransactionConnections,
-		uint64(w.totalConnectionCount)-w.PinnedCursorConnections-w.PinnedTransactionConnections)
+	msg := fmt.Sprintf("%s; maxPoolSize: %d, ", errorMsg, w.maxPoolSize)
+	if pinnedConnections := w.pinnedConnections; pinnedConnections != nil {
+		openConnectionCount := uint64(w.totalConnectionCount) -
+			(*pinnedConnections).cursorConnections -
+			(*pinnedConnections).transactionConnections
+		msg += fmt.Sprintf("connections in use by cursors: %d, connections in use by transactions: %d, connections in use by other operations: %d, ",
+			(*pinnedConnections).cursorConnections,
+			(*pinnedConnections).transactionConnections,
```

Comment:
These explicit dereferences are unnecessary and can be removed.

```suggestion
			pinnedConnections.cursorConnections -
			pinnedConnections.transactionConnections
		msg += fmt.Sprintf("connections in use by cursors: %d, connections in use by transactions: %d, connections in use by other operations: %d, ",
			pinnedConnections.cursorConnections,
			pinnedConnections.transactionConnections,
```